lib/ecom/session.ts

```diff
@@ -43,9 +48,8 @@ export async function initializeEcomSession(request: Request) {
     return { wixEcomTokens, session, shouldUpdateSessionCookie };
 }
 
-export async function initializeEcomApi(request: Request) {
+export async function initializeEcomApiForRequest(request: Request) {
     const { session } = await initializeEcomSession(request);
     const tokens = session.get('wixEcomTokens');
-    const client = createWixClient(tokens);
-    return createApi(client);
+    return tokens ? initializeEcomApiWithTokens(tokens) : initializeEcomApiAnonymous();
```

Comment:
@olehrakwix is it an expected scenario that we won't have tokens here?

Reply:
we are getting tokens from root loader, but if it fails and ErrorBoundary will render a page - we will not have tokens there

Reply:
Should we throw then instead of initializing for anonymous user?

Reply:
I don't think so... in that case we will throw error from ErrorBoundary that will break the whole app I believe.